### ISSUE TITLE
Multiple health check endpoints

### ISSUE THREAD

**Issue by steveoh** [aspnet/Diagnostics#512]
In preview 2 I was able to use 
```cs
      app.UseHealthChecks("/health");
      app.UseHealthChecks("/health/detail", new HealthCheckOptions()
            {
                // This custom writer formats the detailed status as JSON.
                ResponseWriter = WriteResponse,
            });

            app.Run(async (context) =>
            {
                await context.Response.WriteAsync("Go to /health to see the health status");
            });
```
And that would work properly. 

In preview 3 both routes return the default data and the response writer is ignored. You can test this in the CustomWriterStartup.cs file by adding the code above.

What is the proper way to expose more than one health check endpoint? Tags?

**Comment by rynowak** [comment]
I think your issue is the same as https://github.com/aspnet/Diagnostics/issues/511- this is using the `.Map` middleware primitive, so if these middleware were ordered in the opposite way, it would work as you expect. 

 I agree that this is surprising and bad, I think we will have change how this is implemented.

**Comment by steveoh** [comment]
Yes I believe this is a duplicate. I will close this and follow that one.

**Comment by rynowak** [comment]
Fixed for the 2.2.0 release. Thanks for the feedback 👍
